integration-tests/spring-boot-2/spring-boot-2-base/src/test/java/co/elastic/apm/spring/boot/AbstractSpringBootTest.java

```diff
@@ -102,7 +102,9 @@ public void greetingShouldReturnDefaultMessage() throws Exception {
         assertThat(transaction.getContext().getUser().getId()).isEqualTo("id");
         assertThat(transaction.getContext().getUser().getEmail()).isEqualTo("email");
         assertThat(transaction.getContext().getUser().getUsername()).isEqualTo("username");
-        assertThat(transaction.getTraceContext().getServiceName()).isEqualTo("spring-boot-test");
+        // as this test runs in a standalone application and not in a servlet container,
+        // the service.name will not be overwritten for the webapp class loader based on spring.application.name
+        assertThat(transaction.getTraceContext().getServiceName()).isNull();
```

Comment:
We should be able to still support this in the case where the auto-detected value is not read from manifest (see previous comment).